exist-core/src/main/resources/org/exist/xquery/lib/xqsuite/xqsuite.xql

```diff
@@ -905,31 +905,26 @@ declare %private function test:assertTrue($result as item()*) as element(report)
         </report>
 };
 
+declare %private function test:safe-effective-boolean-value ($result as item()*) as xs:boolean {
+    try {
+        boolean($result)
```

Comment:
does this work? the same way? nice....

Reply:
Yes I tested it extensively and also compared it with `not(not($result)`

Reply:
https://gist.github.com/line-o/4f06be97455ec9e1049da5785ab372f9